### ISSUE TITLE
Changing the game installation directory in the preferences doesn't work

### ISSUE THREAD

**Issue by berarma** [sharkwouter/minigalaxy#109]
Hi.

Thanks for this app. I've installed the deb package version 0.9.3 on Debian 10.

I'm trying to change the directory to `/usr/local/games/GOG Games` but it doesn't change. I've deleted the directory created by installing from the deb package in my home then minigalaxy throws an error. If I try to change the directory again it throws another error. In the end I've changed directly the directory in the config file and it worked.

The output of minigalaxy after deleting the default directory in the home dir and then trying to change it in preferences (changing the directory before this had no effects):

```
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3/dist-packages/minigalaxy/ui/library.py", line 45, in __update_library
    self.games = self.__get_installed_games()
  File "/usr/lib/python3/dist-packages/minigalaxy/ui/library.py", line 105, in __get_installed_games
    directories = os.listdir(Config.get("install_dir"))
FileNotFoundError: [Errno 2] No existe el fichero o el directorio: '/home/berarma/GOG Games'

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/minigalaxy/ui/preferences.py", line 113, in save_pressed
    text=_("{} isn't a usable path").format(self.entry_installpath.get_text())
AttributeError: 'Preferences' object has no attribute 'entry_installpath'
```

Thanks.

**Comment by sharkwouter** [comment]
Thanks for reporting. I'll take a look at it.

As a workaround, you can create the directory again manually. Then you should be able to change it without issue. If you change your installation directory while your current one is empty it will delete the old one for you.

**Comment by berarma** [comment]
No, I couldn't change it. It alwas went back to the original directory. I don't know if it has something to do with the directory name being the same but with different path prefixes: `/usr/local/games/GOG Games` and `/home/user/GOG Games`.

That's the reason I tried to tweak things and removing the default directory.

**Comment by sharkwouter** [comment]
I didn't get to it today unfortunately. I will try it, though. Thanks for the additional information!

Something else you could try is editing the configuration file directly or resetting it. The configuration file is in ``~/.config/minigalaxy``. Alternatively you could delete it with ``minigalaxy --reset``.

**Comment by berarma** [comment]
I've reset the configuration and when I try to change the directory in the preferences I get this error:

> Traceback (most recent call last):
>   File "/usr/lib/python3/dist-packages/minigalaxy/ui/preferences.py", line 113, in save_pressed
>     text=_("{} isn't a usable path").format(self.entry_installpath.get_text())
> AttributeError: 'Preferences' object has no attribute 'entry_installpath'

I've edited the config file by hand and at the next start my installed games are found but without icon (the icon was there before changing the directory).

**Comment by sharkwouter** [comment]
This is really strange indeed. It could be an issue with the default configuration, I'll have to check. I'm kinda hoping more people chime in on this, to see if it affects more people.

The icons are a separate issue, though. The icons are stored in a game's directory, but they should be downloaded when missing.

Just to make sure, you haven't run multiple instances of Minigalaxy at the same time, right?

**Comment by berarma** [comment]
> Just to make sure, you haven't run multiple instances of Minigalaxy at the same time, right?

No, I'm sure I didn't.

> The icons are a separate issue, though. The icons are stored in a game's directory, but they should be downloaded when missing.

Yes, they are, but when fixing one issue the other one comes up.

**Comment by berarma** [comment]
The variable `entry_installpath` is used in this line but it isn't assigned anywhere. Why is that?

**Comment by sharkwouter** [comment]
This commit fixes the issue you mentioned in your last comment: https://github.com/sharkwouter/minigalaxy/commit/62741166b37e5753dece6fbc1ce0e067d593ac49

**Comment by sharkwouter** [comment]
I've just tested deleting the installation directory and then restarting Minigalaxy and trying to change the directory. This worked, but I did see an exception at startup and my games didn't load. I would say this is acceptable in this case.

You were running into the issue with the error message because you chose a directory in which Minigalaxy couldn't write. You cannot set the installation directory in a place where you user cannot write.

Thanks for bringing the issue with the error message to my attention
